app/models/submission.rb

```diff
@@ -110,6 +112,12 @@ def status
     end
   end
 
+  def cannot_see_results?(user)
+    (all_tests_passed? || tests_ran?) &&
```

Comment:
Was just thinking, if this shouldn't be in negative form. most other similar kind of methods are asking positive result like visible_to, so should this be can_see_results? just an idea, not sure if it's better.. if this is more clear on calling.